base/core/load.zsh

```diff
@@ -31,7 +31,12 @@ __zplug::core::load::from_cache()
 
         # Plugins with defer-level set
         source "$_zplug_cache[defer_1_plugin]"
-        compinit -d "$ZPLUG_HOME/zcompdump"
+        if [ ${UID} ] && [ -n ${SUDO_USER} ] ; then
```

Comment:
one [ to much in front of SUDO_USER. Variables in {} is in this case unneeded but general good style, this way variables can be chained and/or substituted.